src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProviderRefreshEventSource.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Azure.Core.Diagnostics;
+using System;
+using System.Diagnostics.Tracing;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
+{
+    [EventSource(Name = EventSourceName)]
+    internal class AzureAppConfigurationProviderRefreshEventSource : EventSource
+    {
+        /// <summary>The name to use for the event source.</summary>
+        private const string EventSourceName = "Microsoft-Extensions-Configuration-AzureAppConfiguration-Refresh";
+
+        private const int LogDebugEvent = 1;
+        private const int LogInformationEvent = 2;
+        private const int LogWarningEvent = 3;
+
```

Comment:
The more I read about EventId ([1](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing.eventattribute.eventid?view=net-7.0)), ([2](https://github.com/microsoft/dotnet-samples/blob/master/Microsoft.Diagnostics.Tracing/EventSource/docs/EventSource.md)), ([3](https://github.com/Azure/azure-sdk-for-net/blob/c915bdae1ad9ec44f1435faade3b0b25e4542ded/sdk/core/Azure.Core/src/Diagnostics/AzureCoreEventSource.cs#L14-L34)), the more I feel like "LogDebug", "LogInfo" and "LogWarning" are not really events - they are just log levels. Events are what we are describing in the log message - i.e., each message in the LoggingConstants class should be a separate event. 
For example, "RefreshKeyValueSettingUpdated" should be a unique event called "RefreshKeyValueSettingUpdatedEvent" and should have its own method in this file. 

```
[Event(RefreshKeyValueSettingUpdatedEvent, Level = EventLevel.Informational, Message = "Setting updated. Key:'{0}'")]
public void RefreshKeyValueSettingUpdated(string key)
{
    WriteEvent(RefreshKeyValueSettingUpdatedEvent, key);
}
```

Then the usage would look like this:
```
AzureAppConfigurationProviderRefreshEventSource.Log.RefreshKeyValueSettingUpdated(key);
```
It's also interesting that it might be possible to filter EventSource logs based on the exact EventId, so users dont have to rely on filtering the log message strings if they want to monitor our logs. It does require some changes to how we currently keep track of what needs to be logged inside ExecuteWithFailOverPolicyAsync method. 
@jimmyca15 do you think it's worth the effort?

Reply:
@avanigupta I do agree that the log level should not be confused with the event. If we're not explicitly passing event ids, we should hardcode 0 for the event id. Similar to [built-in usage](https://github.com/aspnet/Logging/blob/2d2f31968229eddb57b6ba3d34696ef366a6c71b/src/Microsoft.Extensions.Logging.Abstractions/LoggerExtensions.cs#L377).

At the moment I wouldn't want to scope-creep and start coming up with unique event ids. We can always update the event ids in the future.

Reply:
Based on the first link mentioned it seems like event id has to be greater than 0 for EventSource. It could be possible to hardcode a different value and pass that to ILogger too.

Reply:
Sure, then can use 1.

Reply:
Sounds good, let's use 1 for all events.